### DISCUSSION TITLE
Implementation custom Cors Provider  using CorsSpi

### DISCUSSION THREAD

**Discussion by ravthiru** [keycloak/keycloak#29547] [Q&A]
Want to provide custom Cors implementation as Keycloak provided default implementation (DefaultCors) does not cater our requirement. There is no way to extend DefaultCors as the constructor is not public.

With latest  keycloak 24 release , CorsSpi allows us to implement  our own provider.

so tried using the code from DefaultCors and wrote custom Cors, by following Server Development documentation https://www.keycloak.org/docs/latest/server_development/#_providers.

Ended up creating following 
a) CustomCorsProvider implementing Cors interface
b) CustomCorsFactory implementing CorsFactory with unique id and order as 1
c) added CustomCorsFactory entry in to META-INF/services/org.keycloak.services.cors.CorsFactory

but i could not see the my custom provider in the Keycloak UI Console and Keycloak is still going with DefaultCors.

Version of Keycloak Used 24.0.3

**Comment by mithi890** [comment]
@ravthiru ,im facing similar issue.how didd you fix it?

**Reply by ravthiru** [reply]
Not able to extened CorsSpi, we ended up create a hack to override **WebOriginsUtils.resolveValidWebOrigins**
